### ISSUE TITLE
[Chrome] Uncaught TypeError: Failed to execute 'getStats' on 'RTCPeerConnection': The callback provided as parameter 1 is not a function.

### ISSUE THREAD

**Issue by OlivierRaulin** [webrtc/testrtc#167]
Hello,

Testing the latest version, I encountered a bug in Chrome.
The commit concerned is the following: https://github.com/webrtc/testrtc/commit/31e48c58c25d599189e463b3dddfe57c6d94690e at line 180 : this.call.pc1.getStats(this.localStream)

It seems from my point of view that a modification was made for Firefox, but Chrome crashes into it because it expects a function as first argument.

**Comment by KaptenJansson** [comment]
I just tested on 47.0.2526.80, 48.0.2564.48 and 49.0.2593.0 and they all work fine for me. What version are you using?

Also FTR we use [adapter.js](https://github.com/webrtc/adapter/blob/master/adapter.js) to shim away browser differences.

**Comment by OlivierRaulin** [comment]
I have been using Chrome 47.0.2526.106 (64-bit) and 48.0.2564.48 beta (64-bit) on Linux and the problem persists.

Maybe I did something wrong, but since I didn't modify that part of code that seems strange to me.

However, as the same code is present on test.webrtc.org and that it does not trigger the error, I guess the mistake is on my side.
I looked at the differences between html sources, and I found nothing that can help me understand why I fall into an error in one case and not another.

I'll close it since I'm not sure this is a bug.

**Comment by KaptenJansson** [comment]
Did you do npm install? It downloads and copies adapter.js to the correct directory.

Also, if you provide more details about what you have done I could be more of help.

**Comment by OlivierRaulin** [comment]
I did the following (as far as I remember) :
- git clone
- npm install
- bower update
- grunt
- grunt build

I was not able to test "Connectivity" and "Throughput" as our test domain is not authorized to call Google Turn servers.

So I updated the source to call our own (and add at the end some more tests that we need), I did again grunt and brunt build, and the tests have been running smoothly until the "Video bandwidth" which fails.

I added specific tests after this last one, which cannot run as this one is failing. Removing the last test does not help.

To summarize updates in the code :
- src/index.html : some html modifications, and an adding of our js test (that can be commented without any effect)
- src/js/call.js : an updateof the URL called for turn credentials, using our own.
- src/js/report.js : I commented the Google Analytics code
- src/js/test{suite,case}name : addings of the test names.

**Comment by KaptenJansson** [comment]
Do you have a camera connected? I assume you have since all other tests work but just want to make sure.

Also, could you add a breakpoint and see what's being passed to the getStats method?

Also, are you using the built version in the out/ folder (generated by grunt build) or the non vulcanized variant (you can point Google app engine to either the root folder or out/)?

**Comment by OlivierRaulin** [comment]
I finally found the origin of the problem.

I forgot that I added a link to our JS library for the final test. Commenting it resolves the problem.
Putting a breakpoint to see the differences, when I navigate to the call object, and into pc1, I see getStats: (selector, successCallback, errorCallback)

However, when I include our library, I got no getStats function into the tree, and when I log it, the console displays [native code]. I guess there's an overwrite taking place somewhere that messes up.

I'll need to get to our developers team to know how we can get around this.
Thanks for your help, it allowed me to go check places I wouldn't have suspected.

**Comment by KaptenJansson** [comment]
No problem, I hope you find a solution ;).
